core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -884,13 +884,23 @@ class AutoTuner(
       getPropertyValue(lookup).getOrElse(DEF_SHUFFLE_PARTITIONS).toInt
     // TODO: Need to look at other metrics for GPU spills (DEBUG mode), and batch sizes metric
     if (isCalculationEnabled(lookup)) {
-      val totalSpilledMetrics = appInfoProvider.getSpilledMetrics.sum
-      if (totalSpilledMetrics > 0) {
+      val totalSpilledMetricsForShuffleStages =
+        appInfoProvider.getSpillMetricsForShuffleStages.unzip._2.sum
+      if (totalSpilledMetricsForShuffleStages > 0) {
         shufflePartitions *= DEF_SHUFFLE_PARTITION_MULTIPLIER
         // Could be memory instead of partitions
         appendOptionalComment(lookup,
           s"'$lookup' should be increased since spilling occurred.")
       }
+
+      val shuffleSkewStages = appInfoProvider.getShuffleSkewStages
+      val shuffleSkewStagesWithSpill = appInfoProvider.getSpillMetricsForShuffleStages.filter {
+        case (stageId, spilledMetrics) => spilledMetrics > 0 && shuffleSkewStages.contains(stageId)
+      }
+      if (shuffleSkewStagesWithSpill.nonEmpty) {
+        appendOptionalComment(lookup, "There is data skew (when task's Shuffle Read Size > 3 * " +
+          s"Avg Stage-level size) in shuffle stages. $lookup recommendation may be less effective.")
```

Comment:
"`bla..bla...$lookup recommendation may be less effective.`"

I do not find this comment easy to understand. "Less effective", what does it mean for the user? Is there a better way to explain this? If we want to say that the recommendation is not that good in the user's case, then why do we recommend it.

Reply:
Updated the message and the logic. Thanks!